motionplan/tpSpaceRRT_test.go

```diff
@@ -303,3 +303,169 @@ func TestTPsmoothing(t *testing.T) {
 	plan = tp.smoothPath(ctx, plan)
 	test.That(t, plan, test.ShouldNotBeNil)
 }
+
+func TestPtgCheckPlan(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	roverGeom, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{10, 10, 10}, "")
+	test.That(t, err, test.ShouldBeNil)
+	geometries := []spatialmath.Geometry{roverGeom}
+	ackermanFrame, err := tpspace.NewPTGFrameFromTurningRadius(
+		"ackframe",
+		logger,
+		300.,
+		testTurnRad,
+		0,
+		geometries,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	goalPos := spatialmath.NewPoseFromPoint(r3.Vector{X: 5000, Y: 0, Z: 0})
+
+	fs := referenceframe.NewEmptyFrameSystem("test")
+	fs.AddFrame(ackermanFrame, fs.World())
+
+	opt := newBasicPlannerOptions(ackermanFrame)
+	opt.SetGoalMetric(ik.NewPositionOnlyMetric(goalPos))
+	opt.DistanceFunc = ik.SquaredNormNoOrientSegmentMetric
+	opt.GoalThreshold = 30.
+
+	test.That(t, err, test.ShouldBeNil)
+	sf, err := newSolverFrame(fs, ackermanFrame.Name(), referenceframe.World, nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	mp, err := newTPSpaceMotionPlanner(ackermanFrame, rand.New(rand.NewSource(42)), logger, opt)
+	test.That(t, err, test.ShouldBeNil)
+	tp, _ := mp.(*tpSpaceRRTMotionPlanner)
+
+	plan, err := tp.plan(context.Background(), goalPos, nil)
+	test.That(t, err, test.ShouldBeNil)
+	planAsInputs := nodesToInputs(plan)
+	test.That(t, err, test.ShouldBeNil)
+	steps := []map[string][]referenceframe.Input{}
+	for _, resultSlice := range planAsInputs {
+		stepMap := sf.sliceToMap(resultSlice)
+		steps = append(steps, stepMap)
+	}
+
+	startPose := spatialmath.NewPoseFromPoint(r3.Vector{0, 0, 0})
+	errorState := startPose
+
+	t.Run("base case - validate plan without obstacles", func(t *testing.T) {
+		err := CheckPlan(ackermanFrame, steps, nil, fs, startPose, errorState)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("obstacles blocking path", func(t *testing.T) {
```

Comment:
Its not worth holding up the PR over, but next time you notice that you have 5 cases that all are checking the same thing you can structure your tests using a for loop as in the example here: https://gobyexample.com/testing.  It makes it more maintainable to not have to write/edit the same thing a bunch of times in a row.  There are a number of examples of tests written this way in the code base if you are looking for examples

Reply:
got rid of redundant test cases